### ISSUE TITLE
Letterbox Function Adding Border

### ISSUE THREAD

**Issue by HunterMorera** [ultralytics/yolov5#8590]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

I wanted to ask why the validation data has a border applied to it before being run through the model, but the training does not. I have noticed that the validation data runs through the letterbox function and a gray border is put around it, is this for any specific reason? It would seem that you would want the training and validation images to be the same when fed through the model.

Currently when you run the same images in training and validation you get different loss curves for box, object, and class losses despite the data being exactly the same. I am not sure if this is some issue in computing the loss. You can see that these loss graphs are different despite the exact same data being passed in for training and validation. Also I am doing pre-augmentation and have turned the augment flag to false, so that wouldn't be the issue. Any help here would be great.

<img width="451" alt="Screen Shot 2022-07-14 at 4 09 01 PM" src="https://user-images.githubusercontent.com/42224519/179278424-83768328-291b-4a4e-afd0-ec215264039f.png">
?

**Comment by glenn-jocher** [comment]
@HunterMorera oh interesting. Did you control for different batch sizes also (val has 2x the batch size of train since it uses less memory due to no gradient).

The half-stride border in val.py helps improve mAP slightly as objects near the edge of an image are a bit harder to detect. You can remove it here by setting pad to 0.0.
https://github.com/ultralytics/yolov5/blob/72a81e7a1c13cd3ae4675037f217d0ed3db9bc20/val.py#L170

**Comment by HunterMorera** [comment]
@glenn-jocher That line does not change the shape of the validation image before it is being passed. I tried printing the pad value and it does not get printed. I am using print(pad,flush=True). I set the else condition here to 0.0 just to be sure and image size still has the border.

Also I changed the validation batch size to match the training and the graphs are still different despite using the same data for training and validation.

My research group at the University of South Florida has expanded the code to work with 10 channel tiff images for medical data. However, we noticed this issue with the training and validation due to our small dataset. Any other suggestions here? Thanks!

**Comment by HunterMorera** [comment]
I would note I am talking about actual validation, not testing.

**Comment by glenn-jocher** [comment]
@HunterMorera the pad variable controls the padding. There's no other source. 

Train and val use the same dataloader so obviously different outputs will be due to different arguments.

**Comment by HunterMorera** [comment]
@glenn-jocher I was able to remove that border by changing line 256 in Train.py to pad=0.0.

**Comment by glenn-jocher** [comment]
@HunterMorera yes exactly that should remove the border.
